test/flag_anomaly.py

```diff
@@ -0,0 +1,126 @@
+import math
+import numpy as np
+import pandas as pd
+import pickle
+
+from tqdm import tqdm
+from ase.geometry import find_mic
+from ase import neighborlist
+from ase.neighborlist import natural_cutoffs
+from ase.io import read, write
+from ase.io.trajectory import Trajectory
+from pymatgen.io.ase import AseAtomsAdaptor
+from pymatgen.core.structure import SiteCollection
+from pymatgen.core.surface import SlabGenerator
+from pymatgen.analysis.local_env import VoronoiNN
+
+def get_connectivity(atoms, multiple=1):
+    """
+    Generate the connectivity of an atoms obj.
+    Args:
+                atoms      An `ase.Atoms` object
+    Returns:
+                matrix     The connectivity matrix of the atoms object.
+    """
+    cutoff = natural_cutoffs(atoms, mult=multiple)
+    neighborList = neighborlist.NeighborList(cutoff, self_interaction=False, bothways=True)
+    neighborList.update(atoms)
+    matrix = neighborlist.get_connectivity_matrix(neighborList.nl).toarray()
+    return matrix
+
+def find_max_movement(init_atoms, final_atoms):
+    '''
+    Given ase.Atoms objects, find the furthest distance that any single atom in
+    a set of atoms traveled (in Angstroms)
+    Args:
+            init_atoms      `ase.Atoms` of the structure in its initial state
+            final_atoms     `ase.Atoms` of the structure in its final state
+    Returns:
+            max_movement    A float indicating the further movement of any single atom
+                            before and after relaxation (in Angstroms)
+    '''
+    # Calculate the distances for each atom
+    distances = final_atoms.positions - init_atoms.positions
+
+    # Reduce the distances in case atoms wrapped around (the minimum image convention)
+    _, movements = find_mic(distances, final_atoms.cell, final_atoms.pbc)
+    max_movement = max(movements)
+    return max_movement
+
+def is_adsorbate_placed_correct(adslab, atoms_tag):
```

Comment:
This isn't called anywhere, did you mean for this to be in DetectAnomaly?

Reply:
I moved it to check_inputs.py because it does make more sense to be there. The check should be done before running DFT.